zetta_utils/training/lightning/regimes/alignment/encoding_coarsener.py

```diff
@@ -60,33 +61,45 @@ def validation_step(self, batch, batch_idx):  # pylint: disable=arguments-differ
         log_row = batch_idx % interval == 0
         sample_name = f"{batch_idx // interval}"
 
+        device = "cuda" if torch.cuda.is_available() else "cpu"
         data_in = batch["data_in"]
+        data_in = data_in.to(device)
```

Comment:
PL is supposed to handle this, right?
Is there no other solution? It's counterintuitive for `validation_step` and `training_step` to have to differ in code this way.